auslib/dockerflow.py

```diff
@@ -26,8 +28,12 @@ def heartbeat():
         """Per the Dockerflow spec:
         Respond to /__heartbeat__ with a HTTP 200 or 5xx on error. This should
         depend on services like the database to also ensure they are healthy."""
-        database_entry_value = heartbeat_database_fn(dbo)
-        return Response(str(database_entry_value), headers={"Cache-Control": "no-cache"})
+        try:
+            database_entry_value = heartbeat_database_fn(dbo)
+            return Response(str(database_entry_value), headers={"Cache-Control": "no-cache"})
+        except Exception as e:
+            logging.exception("Error occured: %s", e.message)
+            return Response(status=502, response=e.message)
```

Comment:
Yup, I was specifically asking about IPs here.

It's hard to guarantee that we're not leaking configuration secrets when blindly returning an Exception message, but SQLAlchemy doesn't print passwords AFAICT, so I think that's very unlikely. I don't think we have any secrets on the public app other than the database password anyways.